### PR TITLE
Make deploy command CI friendly

### PR DESCRIPTION
**Pull request by bergundy**
Please do not review yet, this is not ready.

### CODE REVIEW

File: cli/src/commands/deploy.ts [hunk 1]
```diff
@@ -172,14 +187,24 @@ export default class Deploy extends Command {
     return value === NEW_APPLICATION ? undefined : value;
   }
 
+  public async findApplicationIdByName(appName: string): Promise<string> {
```
Comment (michaeladda) [inline cli/src/commands/deploy.ts]: CR: Use `const { id } = this.lycanClient.getAppByName(appName)`

File: cli/src/commands/deploy.ts [hunk 2]
```diff
@@ -28,7 +31,7 @@ import {
 // * project associated deploys to associated app
 // * project associated and deploy fails shows meaningful message
 //
-// TODO: add --app-id and --env flags to bypass app association mechanism
+// TODO: add --env flag to bypass app association mechanism
```
Comment (michaeladda) [inline cli/src/commands/deploy.ts]: CR: remove the `--env` TODO too

File: cli/src/test/deploy.spec.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    {
+      applicationId: 'abc',
+      defaultEnv: 'default',
+      directory: t.context.projectDir,
+    },
+  ]);
+});
+
+//
+// test('no args not in project dir', async (t) => {
```
Comment (michaeladda) [inline cli/src/test/deploy.spec.ts]: why are these disabled?

File: cli/src/commands/deploy.ts (hunk 2 above)
Comment (bergundy) [reply cli/src/commands/deploy.ts]: To be able to deploy to the non `default` app environment, different from env variables.

File: cli/src/commands/deploy.ts (hunk 1 above)
Comment (bergundy) [reply cli/src/commands/deploy.ts]: Thanks!

File: cli/src/test/snapshots/deploy.spec.ts.md [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+      out: `Building and bundling your app! This may take a few moments, please wait␊
+      ␊
+      > poll@ build PROJECT_DIR␊
+      > echo nothing to do␊
+      ␊
+      nothing to do␊
+      Preparing backend...␊
+      built␊
+      Uploading your assets! This may take a few moments, please wait␊
+      [?25l[36m?[39m [1mSelect target[22m [90m›[39m [90m- Use arrow-keys. Return to submit.[39m␊
```
[the remark this answers is not in the document]
Comment (bergundy) [reply cli/src/test/snapshots/deploy.spec.ts.md]: Good point, I can give it a try

[the remark this answers is not in the document]
Comment (bergundy) [reply cli/src/test/snapshots/deploy.spec.ts.md]: There doesn't seem to be an option in the `prompts` library to turn it off, internally it uses [kleur](https://www.npmjs.com/package/kleur) which supports turning it off but I'd rather not mess around with it for tests.
Filtering it manually doesn't seem ideal either.
Keeping it for now.

**Pull request merged**